source/slang/slang-ast-synthesis.cpp

```diff
@@ -0,0 +1,175 @@
+#include "slang-ast-synthesis.h"
+
+namespace Slang
```

Comment:
It looks like we're doing two separate code synthesis passes for auto-diff as a whole, one on the AST-side and one on the IR-side. 
I'm wondering if it's feasible to synthesize just the function headers (and store any required witness tables), and synthesize the method on the IR side?